### PR TITLE
new readme, new company listing incl logos, new team listing

### PR DESCRIPTION
**Pull request by rvagg**
Complete overhaul. The main highlights are:

* Company listing including blurbs and logos, see the rendered version at <https://github.com/nodejs/build/blob/new-readme/README.md>. The main aim here is to recognise and thank these companies for contributing, with a secondary aim to provide additional incentive for other companies to contribute (i.e. we'll give you a PR boost if you contribute). I've separated them into two tiers because DO and Rackspace really do hold us up and need kudos for doing so.
* New team listing, based on activity and attendance. I hope I'm not offending by adjusting this, please speak up if you think I've got this wrong, it's simply based on what I _feel_ to be the current list of active participants in the work we do here but I could easily be overlooking people.

I also propose that if this PR is merged that the team listed become the @nodejs/build team, updated to remove inactive members and add newly active members.

The team that I've listed in this PR atm is:

* Johan Bergström @jbergstroem
* João Reis @joaocgreis
* Rod Vagg @rvagg
* Alexis Campailla @orangemocha
* Michael Dawson @mhdawson
* Hans Kristian Flaatten @Starefossen
* Julien Gilli @misterdjules
* Rich Trott @trott
* Myles Borins @thealphanerd
* Ryan Graham @rmg

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,117 +1,69 @@
...
 
-All commits and pull requests will be submitted for compilation and execution of the **test-simple** test suite on the following platforms:
+![Tier 2 Infrastructure Providers](./provider-logos/tier-2-providers.png)
 
-  * Containerized 64bit image of the current oldest Ubuntu LTS Release (10.04, Lucid Lynx)
-  * Containerized 64bit image of the current latest Ubuntu LTS Release (14.04, Trusty Tahr)
+**[Microsoft Azure](https://azure.microsoft.com/en-us/)**, a cloud services platform, provide Windows test infrastructure for the Node.js CI system.
 
-### Configurations: Core & easy
+**[Joyent](https://www.joyent.com/)**, a public/private cloud infrastructure company, provide SmartOS test and build resources for the Node.js CI system.
```
Comment (jbergstroem) [inline README.md]: Nitpick: we run a linter here -- perhaps mention that they provide general resources as well.

File: README.md [hunk 2]
```diff
@@ -1,117 +1,69 @@
...
 
-Additionally, all commits by core contributors will be submitted for compilation and execution of the **test-all** test suite on the following platforms:
+**[IBM](https://www.ibm.com/)**, via their cloud company, [SoftLayer](https://www.softlayer.com/) and the [Oregon State University Open Source Lab](https://osuosl.org/services/powerdev) provide PPC-based test and build infrastructure and other key hardware for testing and benchmarking for the Node.js project's CI system.
```
Comment (jbergstroem) [inline README.md]: I would split this up into softlayer and osuosl. Albeit both might be provided through IBM they have different agendas and both deserve their own spot.

softlayer provides us with general vm's for testing (and improved redundancy/latency) as well as baremetal for benchmarking. osuosl provides our ppc stack.

**Comment by jbergstroem** [comment]
Great to finally see our `README.md` getting some treatment :clap:  -- comments inline. Also, how about we ask the members of the suggested working group if they (still) want to be part of it?

**Comment by Starefossen** [comment]
Looks very good, Rod! Nice work :this mbsup:

File: README.md [hunk 3]
```diff
@@ -1,117 +1,69 @@
-Node Test and Build Infrastructure
-==================================
+Node.js Build Working Group
+===========================
 
-Chat with us! We use IRC: [#node-build at freenode](irc://irc.freenode.net/node-build).
+Chat with us! We use IRC: [#node-build at Freenode](irc://irc.freenode.net/node-build)
 
 Wut?
 ----
```
Comment (jbergstroem) [inline README.md]: How about adding a sentience about the build group?

`The node.js build working group maintains and controls infrastructure used for continuous integration, releases, benchmarks and more. Our mission is to provide node.js with a solid testing infrastructure that aims to improve the quality of the software itself by targeting correctness, speed or cross-compatibility issues.`

File: README.md [hunk 4]
```diff
@@ -1,117 +1,69 @@
...
+* A summary of build and test jobs can be found at: <https://ci.nodejs.org>
+* A listing of connected servers for testing, building and benchmarking can be found at: <http://ci.nodejs.org/computer/>
 
 
 People
 ------
 
-* Rod Vagg [@rvagg](https://github.com/rvagg)
-* Ben Noordhuis [@bnoordhuis](https://github.com/bnoordhuis)
-* Ryan Graham [@rmg](https://github.com/rmg)
```
Comment (rmg) [inline README.md]: I still consider myself a member and I'd like to remain on the list, if there aren't any objections. Regular involvement is one of the sources of light at the end of the acquisition/integration tunnel I've been looking forward to.

**Comment by orangemocha** [comment]
Thanks for doing this, @rvagg ! 

One nit about Microsoft sponsorship: the sponsor is actually "Microsoft". "Microsoft Azure" is the platform where some (ok, all except for the serial port machine) of those hardware resources are hosted.

LGTM otherwise.

File: README.md (hunk 4 above)
Comment (rvagg) [inline README.md]: Sorry @rmg, I should have asked! Will put you back on.

@bnoordhuis, @wblankenship, @kenperkins, @geek: I took you all off due to perceived (on my part) inactivity. Can you clarify your availability and interest in continuing to be involved and listed as members of the build team?

**Comment by rvagg** [comment]
@orangemocha can you check with someone at Microsoft re what they'd like represented? I've got the impression from folks there that pushing the Azure brand is pretty important at this stage. Happy to comply with whatever you/they want to go with.

**Comment by orangemocha** [comment]
Ok, @rvagg , I am double checking. Thanks!

**Comment by orangemocha** [comment]
One more nit: all verbs where the subject is a company name appear conjugated in the 3rd plural person "provide" "donate". Shouldn't it be "provides" "donates"? (disclaimer: I am not a native English speaker)

**Comment by mhdawson** [comment]
Could you change

IBM, via their cloud company, SoftLayer and the Oregon State University Open Source Lab provide PPC-based test and build infrastructure and other key hardware for testing and benchmarking for the Node.js project's CI system.

to:

IBM, via their cloud company, SoftLayer and the Oregon State University Open Source Lab provide PPC-based test and build infrastructure and other key x86 hardware for testing and benchmarking for the Node.js project's CI system.

I don't want it to sound like we only provide PPC resources.

**Comment by mhdawson** [comment]
Just read @jbergstroem  earlier comment. Splitting it up would also work

File: README.md (hunk 4 above)
Comment (rmg) [inline README.md]: :+1:

**Comment by Trott** [comment]
> One more nit: all verbs where the subject is a company name appear conjugated in the 3rd plural person "provide" "donate". Shouldn't it be "provides" "donates"? (disclaimer: I am not a native English speaker)

In US English, it would be "IBM provides..." etc. In UK English, company names (and band names and all sorts of things that represent collections of more than one person) are *collective nouns* and treated as a plural noun. "IBM provide..."

So either could be correct.

It seems that Node.js has standardized on US English. We seem to *favor* US English rather than *favour* UK English. So I would be mildly inclined to go with @orangemocha's suggestion. But again, either could be correct.

**Comment by rvagg** [comment]
I think I've addressed all of the comments raised here so far, please review latest commit

**Comment by rvagg** [comment]
Added a new section on community donations, listing all of the donations and donaters that we've had so far, including hardware that is not in active service.

**Comment by jbergstroem** [comment]
Looking great, Rod. LGTM :shipit:

**Comment by rmg** [comment]
LGTM

**Comment by joshgav** [comment]
Representing Microsoft's view, I agree with @orangemocha that our company name should be Microsoft, not Microsoft Azure. Here's what I propose for the Microsoft line:

> **[Microsoft](https://www.microsoft.com/)** provides Windows-related test infrastructure on [Azure](https://azure.microsoft.com) for the Node.js CI system.

Could we take the logo from our [logo page](https://www.microsoft.com/en-us/legal/intellectualproperty/trademarks/usage/logo.aspx) (as below) for the image? 

Microsoft Logo

---

Somewhat relatedly, do we have a quantifiable differentiation between "1st tier" and "2nd tier"? It's something that might cause controversy so would be nice to be crisp. Also, perhaps "gold" and "silver" or any sequence that doesn't number sponsors might be more politic :)

File: README.md [hunk 5]
```diff
@@ -1,117 +1,99 @@
...
+Chat with us! We use IRC: [#node-build at Freenode](irc://irc.freenode.net/node-build)
 
 Wut?
 ----
 
-This repository contains information used to set up and maintain the **io.js** and **libuv** CI infrastructure. It is intended to be open and transparent, if you see any relevant information missing please open an issue.
+The Node.js Build Working Group maintains and controls infrastructure used for continuous integration (CI), releases, benchmarks, web hosting (of [nodejs.org](https://nodejs.org/) and other Node.js web properties) and more.
 
-io.js and libuv are tested on a specific set of hardware / operating system / configuration combinations. Commits to the repository are tested on the full set while pull requests to the Node.js and libuv projects from non-core contributors are tested on a smaller, more secure subset. Build and test output is collected and success or fail status is reported back to GitHub.
+Our mission is to provide Node.js Foundation projects with solid computing infrastructure in order improve the quality of the software itself by targeting correctness, speed and compatibility and to ensure streamlined delivery of binaries and source code to end-users.
```
Comment (joshgav) [inline README.md]: edit: in order *to* improve

**Comment by mikeal** [comment]
Some companies are particular about how/where/when their logo is used. If we get a complaint please forward it to me right away and I'll figure out what the foundation need to do in order to get approval.

**Comment by mhdawson** [comment]
lgtm

**Comment by mhdawson** [comment]
One thing we discussed in the last WG meeting was to define a concrete definition for what contributes makes Tier 1 (# cpus, memory etc.)

**Comment by rvagg** [comment]
A concrete definition is going to be hard for two main reasons:

1. Some providers don't want us to publicise how much we are using their stuff, for various reasons
2. Making a comparison between resources of the various providers is nearly impossible due to huge variations in what resources are available and used.

The definition for now is based on a rough calculation of **$$** spent if we were paying for these, and by far DigitalOcean and Rackspace lead everyone else in how much they are going out of their way to support the project. So IMO, putting them in a special category at the top is both appropriate and deserved. That said, we have contenders that have the potential to reach up in to Tier 1 for the same reason. SoftLayer, Joyent and Azure could be ramped up to reach similar levels.

I haven't done the spend-analysis for a few months, the Foundation Board asked for it a while back to understand the risks associated with having in-kind offerings. My suggestion would be for the core Build group to privately maintain this analysis, updated with latest usage across providers and when we have the numbers is should be very clear who is in Tier 1, _or_ whether the blurring between the tiers is enough to justify adding another or removing the designations all together.

Objections from Tier 2 providers about having the tiered designation suggests to me that it works as an incentive. I'm more than happy to have providers compete to be in the top tier and be thanked more loudly than others, this whole PR is largely about incentive construction.

**Comment by rvagg** [comment]
I've just uploaded the spreadsheet I made back in August into Google Docs and shared it with @jbergstroem, @joaocgreis, @orangemocha and @mhdawson. It's very out of date and needs a lot of work to become accurate again. Let's try and keep this maintained (but the numbers private to respect the wishes of the various companies please).

**Comment by rvagg** [comment]
If only companies offered downloadable versions of their logos while telling people _how_ to use their logos ... @joshgav can you point me to a higher res version of the Microsoft logos that I can use? https://c.s-microsoft.com/en-us/CMSImages/imgOne.jpg?version=2089bd70-681c-1e6e-0b1a-680002dd31d6 is not terrible but surely there's a vector or high-res version of individual files instead of something I have to cut up manually?

**Comment by rmg** [comment]
http://news.microsoft.com/microsoft-news-center-photos/msft_logo_rgb_c-gray/ should do the trick..

**Comment by rvagg** [comment]
thanks @rmg, although that brand use page says to use it on white or blue, this is grey!

**Comment by rvagg** [comment]
oh, nmind, that grey must be the transparency colour

**Comment by rvagg** [comment]
Updated the Microsoft logo and text, latest can be seen @ https://github.com/nodejs/build/blob/234e3dadd28cafffef395e2dab37d776d18ef67e/README.md

How firm are the objections to the tiers? Enough to hold this up or can we go ahead with this and fix it up later?

**Comment by jbergstroem** [comment]
Regarding tiers: whatever way we currently define it, digitalocean and rackspace are most used. I'm happy to merge without a clear definition of tiers and improve as we go.

**Comment by orangemocha** [comment]
> I've just uploaded the spreadsheet I made back in August into Google Docs

Thanks @rvagg , I will update the spreadsheet with the Azure spend.

It's not a problem to merge as is and refine things later. Once we do, I think we might want to set a time period over which to compute a sponsor's contribution (in $). 1 month is probably too short as these numbers can fluctuate. If we were to consider contributions from all time, and even before the Foundation, Joyent and Microsoft would definitely have to be in the top tier. I would suggest to use a "last 12 month" period and define the tiers in terms of $ thresholds.

**Comment by mhdawson** [comment]
I'm fine with merging as is.

**Comment by joshgav** [comment]
LGTM. Thanks!

**Comment by rvagg** [comment]
> If we were to consider contributions from all time, and even before the Foundation, Joyent and Microsoft would definitely have to be in the top tier.

Maybe, you have to consider that the resources we are using across the board are on a totally different scale than were ever maintained for jenkins.nodejs.org. I'd not be surprised if total spend in the last 12 months completely eclipsed any previous total spend on infra by the project.

Personally I'd rather time-limit it or even make it a snapshot thing cause calculating spend over spans of time is way more difficult and would require more dillegence than any of us can afford. So perhaps it's a quarterly, or 1/2 yearly thing we do.

Let's aim to have the spreadsheet updated by the next Build WG meeting we have and we'll come up with a strategy that properly recognises the major contributors and puts incentives in place that encourage new contributing providers and existing providers to level-up how they contribute.

**Comment by jbergstroem** [comment]
Yeeeees :tada:

**Comment by rvagg** [comment]
:boom:

improvements via PR please

**Comment by orangemocha** [comment]
:+1:

**Pull request closed without merging**